test/integration/controller/core/workload_controller_test.go

```diff
@@ -269,10 +269,11 @@ var _ = ginkgo.Describe("Workload controller", ginkgo.Ordered, ginkgo.ContinueOn
 					gomega.Expect(k8sClient.Get(ctx, wlKey, &createdWl)).To(gomega.Succeed())
 					return apimeta.FindStatusCondition(createdWl.Status.Conditions, kueue.WorkloadFinished)
 				}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(&metav1.Condition{
-					Type:    kueue.WorkloadFinished,
-					Status:  metav1.ConditionTrue,
-					Reason:  "AdmissionChecksRejected",
-					Message: "Admission checks [check1] are rejected",
+					Type:               kueue.WorkloadFinished,
+					Status:             metav1.ConditionTrue,
+					Reason:             "AdmissionChecksRejected",
+					Message:            "Admission checks [check1] are rejected",
+					ObservedGeneration: createdWl.Generation,
```

Comment:
Lets ignore the generation in the integration tests, and check the seating in the unit tests.